This patch is based on the upstream commit described below, adapted for
use in the Debian package by Peter Michael Green.

commit e1ec9e63f9e9032b3acaaf411be10a2fadba78a0
Author: Chojan Shang <psiace@outlook.com>
Date:   Mon Aug 23 21:25:38 2021 +0800

    Use seed_from_u64
    
    Signed-off-by: Chojan Shang <psiace@outlook.com>

only in patch2:
Index: hashbrown/src/map.rs
===================================================================
--- hashbrown.orig/src/map.rs
+++ hashbrown/src/map.rs
@@ -4271,11 +4271,8 @@ mod test_map {
         let mut m = HashMap::new();
 
         let mut rng = {
-            let seed = [
-                1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, // and again
-                1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
-            ];
-            SmallRng::from_seed(seed)
+            let seed = u64::from_le_bytes(*b"testseed");
+            SmallRng::seed_from_u64(seed)
         };
 
         // Populate the map with some items.
@@ -4584,11 +4581,8 @@ mod test_map {
         let mut m = HashMap::new();
 
         let mut rng = {
-            let seed = [
-                1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, // and again
-                1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
-            ];
-            SmallRng::from_seed(seed)
+            let seed = u64::from_le_bytes(*b"testseed");
+            SmallRng::seed_from_u64(seed)
         };
 
         // Populate the map with some items.
Index: hashbrown/tests/set.rs
===================================================================
--- hashbrown.orig/tests/set.rs
+++ hashbrown/tests/set.rs
@@ -8,12 +8,9 @@ fn test_hashset_insert_remove() {
     let mut m: HashSet<Vec<char>> = HashSet::new();
     //let num: u32 = 4096;
     //let tx: Vec<Vec<u8>> = (0..num).map(|i| (i..(16 + i)).collect()).collect();
-    let seed: [u8; 32] = [
-        130, 220, 246, 217, 111, 124, 221, 189, 190, 234, 121, 93, 67, 95, 100, 43, // again
-        130, 220, 246, 217, 111, 124, 221, 189, 190, 234, 121, 93, 67, 95, 100, 43,
-    ];
+    let seed = u64::from_le_bytes(*b"testseed");
 
-    let rng = &mut SmallRng::from_seed(seed);
+    let rng = &mut SmallRng::seed_from_u64(seed);
     let tx: Vec<Vec<char>> = (0..4096)
         .map(|_| {
             rng.sample_iter(&Alphanumeric)
